### PR TITLE
Add pratio and maxdim

### LINKED ISSUE #88: Add `pratio` option to EigenAnalysis
Currently we need to specify the number of output dimensions, but an useful option available in MultivariateStats.jl is `pratio` which determines the number of dimensions to preserve in order to account for a ratio of the total variance.

### PR DESCRIPTION
**Pull request by eliascarv**
This PR adds `pratio` option to `EigenAnalysis` and change `ndim` to `maxdim`.
I made some decisions to this implementation:
* Make `pratio` and `maxdim` kwargs to give the user option to choose one of these, both or neither.
* Change `ndim` to `maxdim` to make sense with `pratio` implementation.

closes #88

### CODE REVIEW

File: src/transforms/eigenanalysis.jl [hunk 1]
```diff
@@ -37,21 +37,23 @@ EigenAnalysis(:V, 2)
...
-  function EigenAnalysis(proj, ndim=nothing)
+  function EigenAnalysis(proj, maxdim, pratio)
     @assert proj ∈ (:V, :VD, :VDV) "Invalid projection."
-    new(proj, ndim)
+    @assert 0 ≤ pratio ≤ 1 "Invalid ratio."
+    new(proj, maxdim, pratio)
   end
 end
 
+EigenAnalysis(proj; maxdim=nothing, pratio=0.99) =
```
Comment (juliohm) [inline src/transforms/eigenanalysis.jl:50]: Why not default 1?

**Review by juliohm — CHANGES_REQUESTED** [review]
Hi @eliascarv I can give a more detailed review after 17hs BRT today. I wonder if we should preserve the semantic of ndim and use ndim and pratio as mutually exclusive options. That is what I had in mind originally.

Let's chat over Zulip to align detail later today.

File: src/transforms/eigenanalysis.jl (hunk 1 above)
Comment (eliascarv) [reply src/transforms/eigenanalysis.jl:50]: Because it's a statistical convention.

File: src/transforms/eigenanalysis.jl [hunk 2]
```diff
@@ -3,10 +3,11 @@
 # ------------------------------------------------------------------
 
 """
-    EigenAnalysis(proj, ndim=nothing)
+    EigenAnalysis(proj; maxdim=nothing, pratio=1.0)
 
 The eigenanalysis of the covariance with a given projection `proj`.
-The number of dimensions of the output is defined by the `ndim` argument.
+`maxdim` keyword argument defines the maximum number of dimensions of the output.
+`pratio` is the percentage ratio of the variances that will be preserved.
```
Comment (juliohm) [inline src/transforms/eigenanalysis.jl:10]:
```suggestion
Optionally specify the maximum number of dimensions in the output `maxdim`
and the percentage of variance to retain `pratio`. Default to all dimensions of
the input.
```

File: src/transforms/eigenanalysis.jl [hunk 3]
```diff
@@ -32,26 +33,30 @@ for more details about these three variants of eigenanalysis.
 EigenAnalysis(:V)
 EigenAnalysis(:VD)
 EigenAnalysis(:VDV)
-EigenAnalysis(:V, 2)
+EigenAnalysis(:V, maxdim=3)
+EigenAnalysis(:VD, pratio=0.89)
+EigenAnalysis(:VDV, maxdim=3, pratio=0.89)
```
Comment (juliohm) [inline src/transforms/eigenanalysis.jl:38]:
```suggestion
EigenAnalysis(:VD, pratio=0.99)
EigenAnalysis(:VDV, maxdim=3, pratio=0.99)
```

**Review by juliohm — CHANGES_REQUESTED** [review]
Review attached.

**Review by juliohm — APPROVED** [review]

**Comment by juliohm** [comment]
Amazing PR as usual @eliascarv , super clean and professional coding 💯 Let's just wait for the tests to pass.

**Pull request merged**